packages/angular/build/src/utils/index-file/auto-csp_spec.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import { autoCsp } from './auto-csp';
+
+describe('auto-csp', () => {
+  it('should rewrite a single inline script', async () => {
+    const result = await autoCsp(`
+      <html>
+        <head>
+        </head>
+        <body>
+          <script>console.log('foo');</script>
+          <div>Some text </div>
+        </body>
+      </html>
+    `);
+
+    expect(result).toContain(
+      `<meta http-equiv="Content-Security-Policy" content="script-src 'strict-dynamic' 'sha256-1kOLrDKT3TBiHLcnxiGsc7HF/lyVJKLhoZDSn0UwCfo=' https: 'unsafe-inline';object-src 'none';base-uri 'self';">`,
+    );
+  });
+
+  it('should rewrite a single source script', async () => {
+    const result = await autoCsp(`
+      <html>
+        <head>
+        </head>
+        <body>
+          <script src="./main.js"></script>
+          <div>Some text </div>
+        </body>
+      </html>
+    `);
+
+    expect(result).toContain(
+      `<meta http-equiv="Content-Security-Policy" content="script-src 'strict-dynamic' 'sha256-lo8BxmGTvJc91EDqVJtKZxnRIqW9+qxQjfaPHuteg74=' https: 'unsafe-inline';object-src 'none';base-uri 'self';">`,
+    );
+    expect(result).toContain(`var scripts = [['./main.js',undefined, false]];`);
+  });
+
+  it('should rewrite a single source script in place', async () => {
+    const result = await autoCsp(`
+      <html>
+        <head>
+        </head>
+        <body>
+          <div>Some text</div>
+          <script src="./main.js"></script>
+        </body>
+      </html>
+    `);
+
+    expect(result).toContain(
+      `<meta http-equiv="Content-Security-Policy" content="script-src 'strict-dynamic' 'sha256-lo8BxmGTvJc91EDqVJtKZxnRIqW9+qxQjfaPHuteg74=' https: 'unsafe-inline';object-src 'none';base-uri 'self';">`,
+    );
+    // Our loader script appears after the HTML text content.
+    expect(result).toMatch(
+      /Some text<\/div>\s*<script>\s*var scripts = \[\['.\/main.js',undefined, false\]\];/,
+    );
+  });
+
+  it('should rewrite a multiple source scripts with attributes', async () => {
+    const result = await autoCsp(`
+      <html>
+        <head>
+        </head>
+        <body>
+          <script src="./main1.js"></script>
+          <script async src="./main2.js"></script>
+          <script type="module" async src="./main3.js"></script>
+          <div>Some text </div>
+        </body>
+      </html>
+    `);
+
+    expect(result).toContain(
+      `<meta http-equiv="Content-Security-Policy" content="script-src 'strict-dynamic' 'sha256-vkcRYaaRLTkuunplxAyjuivyXpK+pbbEfJB92l8u+aY=' https: 'unsafe-inline';object-src 'none';base-uri 'self';">`,
+    );
+    expect(result).toContain(
+      `var scripts = [['./main1.js',undefined, false],['./main2.js',undefined, false],['./main3.js','module', false]];`,
+    );
```

Comment:
Consider: Maybe verify that we only have one `<script>` tag in the output? That way we ensure they were batched together as expected.

Reply:
Tested!